StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1134UnitTests.cs

```diff
@@ -409,5 +409,35 @@ public class TestClass<[Test(""Test1"")][Test(""Test2"")]T>
 
             await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Verifies that passing an invalid member syntax into the codefix will not change the code.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [WorkItem(2894, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2894")]
+        public async Task VerifyInvalidMemberSyntaxInCodeFixAsync()
+        {
+            string testCode = @"class Program
+{
+    static void Main(string[] args)
+    {
+        {
+        }[;]
+    }
+}
+";
+
+            DiagnosticResult[] expected =
+            {
+                DiagnosticResult.CompilerError("CS1513").WithLocation(6, 10),
+                Diagnostic().WithLocation(6, 10),
```

Comment:
❓ Would it be better to avoid reporting a diagnostic in this case?

Reply:
I was wondering that myself as well. On the one hand, there is enough syntax available to allow the analyzer to report a diagnostic (without crashing on a wrong syntax), but on the other hand there is a problem there when registering the code fix.

I think this is an edge case and as the code fix was easy to repair, I opted for that.